packages/compass-components/src/components/signal-popover.tsx

```diff
@@ -549,12 +552,20 @@ const SignalPopover: React.FunctionComponent<SignalPopoverProps> = ({
             description="Across Compass, you may now see icons like this to clue you in on potential areas of improvement for your data."
             buttonText="See insights in action"
             onPrimaryButtonClick={() => {
-              triggerRef.current?.click();
+              // Because the guide cue is currently in inactive state when this
+              // button is clicked, the popover position can be calculated
+              // incorrectly because the expand animation will be triggered at
+              // the same time as popover show animation. To work around that,
+              // we will first manually trigger, wait for the transition
+              // duration, and only then will click the trigger to open the
+              // popup
+              setHovered(true);
+              setTimeout(() => {
+                rafraf(() => {
+                  triggerRef.current?.click();
+                });
+              }, TRANSITION_DURATION_MS);
             }}
-            // So that the insight badge can animate without messing the tooltip
-            // position
-            tooltipAlign="right"
```

Comment:
This sort of worked, but not for all the cases where the insights badge could be first seen by the user